utils/zerovec/src/map/hashmap.rs

```diff
@@ -0,0 +1,270 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use crate::ule::AsULE;
+use crate::ZeroVec;
+use ahash::AHasher;
+use alloc::borrow::Borrow;
+use alloc::vec::{from_elem, Vec};
+use core::hash::{Hash, Hasher};
+
+#[inline]
+fn create_hasher_with_seed(seed: u128) -> AHasher {
+    AHasher::new_with_keys(seed, 0xaabbccdd)
+}
+
+#[inline]
+fn compute_hash<K: Hash>(seed: u32, k: K, m: usize) -> u32 {
+    let mut hasher = create_hasher_with_seed(seed.into());
+    k.hash(&mut hasher);
+    (hasher.finish() % m as u64) as u32
+}
+
+#[derive(Debug, Clone)]
+pub struct HashIndex<'a> {
+    displacements: ZeroVec<'a, u32>,
+    reverse_mapping: ZeroVec<'a, u32>,
```

Comment:
Suggestion: use `FlexZeroVec` so that you can store and read `usize` values (and also save some space). It has a small lookup penalty, but this should be minor relative to the hash calculation.